src/test/java/com/epam/steps/admin/AddTeacherToSubjectSteps.java

```diff
@@ -0,0 +1,106 @@
+package com.epam.steps.admin;
+
+import com.epam.helpers.SharedTestData;
+import com.epam.pages.main.SubjectPage;
+import com.epam.pages.popup.SubjectPopup;
+import com.epam.steps.BaseSteps;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AddTeacherToSubjectSteps extends BaseSteps {
+    private SubjectPage subjectPage;
+    private SubjectPopup subjectPopup;
+
+    @Before
+    public void initPages() {
+        subjectPage = new SubjectPage();
+        subjectPopup = new SubjectPopup();
+    }
+
+    @And("Click on the {} subject in the list")
+    public void clickOnTheSubjectInTheList(String subjectName) {
+        subjectPage.clickOnSubjectFromTheList(subjectName);
+    }
+
+    @And("Click on 'Add' button and open popup")
+    public void clickOnAddButtonAndOpenPopup() {
+        subjectPage.clickOnAddButton();
+    }
+
+    @Then("Check all elements are present in subject dashboard teachers section")
+    public void checkAllElementsArePresentInSubjectDashboardTeachersSection() {
+        assertThat(subjectPopup.checkUIofAddTeachersPopup())
+                .withFailMessage("Elements are not displayed")
+                .isTrue();
+    }
+
+    @And("Select teachers section in subject dashboard")
+    public void selectTeachersSectionInSubjectDashboard() {
+        subjectPage.clickOnTeachersSection();
+    }
+
+    @And("Teacher for subject is not displayed on the list")
+    public void teacherForSubjectIsNotDisplayedOnTheList() {
+        assertThat(subjectPage.getTeacherListSize()).isEqualTo(SharedTestData.getListSize());
+    }
+
+    @And("Teacher for subject is displayed on the list")
+    public void teacherForSubjectIsDisplayedOnTheList() {
+        assertThat(subjectPage.getTeacherListSize()).isGreaterThan(SharedTestData.getListSize());
```

Comment:
Please write a fail error message for this assertion.

Reply:
done